### PR TITLE
fix: Depend on HEAD indirectly to support worktrees

### PR DESCRIPTION
**Pull request by Bownairo**

### CODE REVIEW

**Review by frankdavid — APPROVED** [review]

File: bazel/workspace_status.sh [hunk 1]
```diff
@@ -33,5 +33,8 @@ elif [[ -n "${HOSTUSER:-}" ]]; then
     echo "USER ${HOSTUSER}"
 fi
 
+# Track HEAD outside of the git directory, to support worktrees.
+git rev-parse --verify HEAD >"${WORKSPACE_ROOT}/git-head.txt"
```
Comment (nmattia) [inline bazel/workspace_status.sh:37]: I'm a bit hesitant to write files from the workspace status. Is the issue that `.git/index` does not work when using git worktrees?

Comment (Bownairo) [reply bazel/workspace_status.sh:37]: That's right. In a worktree, `.git` is a text file that points back at the main tree's `.git` dir. https://git-scm.com/docs/git-worktree#_details

A small nit is that `HEAD` can change without changing the index, for example with an empty commit, but we can fix that without adding to `workspace_status.sh`.


I don't know how common worktrees are. I should be able to work around this in my own environment if we'd rather wait until there's a better stamping solution.

Comment (nmattia) [reply bazel/workspace_status.sh:37]: > I should be able to work around this in my own environment if we'd rather wait until there's a better stamping solution.

Then let's do that if you don't mind! Or at least give it a try, and if it doesn't work out we can always revisit. I'm hoping we'll have time to look at this stamping Soon™....

Comment (Bownairo) [reply bazel/workspace_status.sh:37]: Sounds good!

Comment (nmattia) [reply bazel/workspace_status.sh:37]: @Bownairo does [this](https://github.com/dfinity/ic/pull/2765) fix the issue in worktrees?

Comment (Bownairo) [reply bazel/workspace_status.sh:37]: It does!

For history, I worked around this by replacing the `.git` file with a symlink to the worktree subdir in the main `.git` folder, and had not run into any issues (yet 🙂).

**Pull request closed without merging**